artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/Main.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.server.embedded;
+
+import java.io.File;
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.CountDownLatch;
+
+import org.apache.activemq.artemis.core.config.FileDeploymentManager;
+import org.apache.activemq.artemis.core.config.impl.ConfigurationImpl;
+import org.apache.activemq.artemis.core.config.impl.FileConfiguration;
+import org.apache.activemq.artemis.core.config.impl.LegacyJMSConfiguration;
+import org.apache.activemq.artemis.core.server.ActivateCallback;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Main {
+
+   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+   private static String workDir = "/app";
+   private static volatile EmbeddedActiveMQ embeddedServer;
+
+   public static void main(String[] args) throws Exception {
+
+      if (args.length == 1) {
+         workDir = args[0];
+         logger.debug("User supplied work dir {}", workDir);
+      }
+
+      String propertiesConfigPath = "/config/," + workDir + "/etc/";
+      if (args.length == 2) {
+         propertiesConfigPath = args[1];
+         logger.debug("User supplied properties config path {}", propertiesConfigPath);
+      }
+
+      FileConfiguration configuration = new FileConfiguration();
+
+      String dataDir = workDir + "/data";
+      configureDataDirectory(configuration, dataDir);
+
+      File bringYourOwnXml = new File(workDir + "/etc/broker.xml");
+      if (bringYourOwnXml.exists()) {
+         logger.debug("byo config found {}", bringYourOwnXml);
+         configuration = loadFromXmlFile(bringYourOwnXml, configuration);
+      }
```

Comment:
I think that would stance be fine if it were at all obvious the disparity was there, and what the default behaviour in any case actually was, and there were jsut a few settings someone had to look into. It really isnt obvious its there, or what the default is, and there are lots of settings, people will likely not know to configure (or wont bother if they do).

I know thatI couldnt could tell you what the behaviour of a lot of these things is just now in a given situation. I'm pretty sure many a regular user couldnt either. If only because the properties themselves also arent even documented yet, though there is an open PR for that...which I recall also commenting on saying that it really didnt make the defaults behaviours very clear, listing default values that often wouldnt be the effective default.

At the very least it feels like this 'different behaviours'..behaviour...should be documented in the readme and byoc example, to clearly convey the general disparities that will occur. I think a user will be quite surprised that a new broker from the CLI and a new broker from this image behave differently out the box in any way except the auth stuff that the baviour is stated for.

Reply:
I think some surprise is expected, because there is so little here. Just a bare java app that contains a broker main line (that happens to have hardcoded defaults). It is a start point to externalise your configuration and actual have the opportunity to lock down your chosen configuration.
It intentionally does not use artemis create or the xml config defaults that can change version to version. It is different exactly for that reason.

Reply:
I have added a default to the Intent: 
https://github.com/apache/activemq-artemis/pull/4297/files#diff-3e9a042c763f10a330adde9100e40b834b67abd3392bce668676fa36132b243aR7

to point out that you get hard coded config defaults. I think that will help drive expectations.

Reply:
I dont think people should be expecting to be surprised by a 'default' broker image, behaving completely differently than a regular 'default' broker instance.  Little things like auto-deletion being enabled when it isnt for a regular broker or the other image. That new persist-redelivery thing Clebert is working on now. Page counters? Etc etc etc. 
as it still inst particularly clear from that update, plus the properties to change the config are as yet undocumented, and finally when we have other broker images that do behave the exactly same as a regular broker instance.

I would at least add something far clearer about the behaviour, like: "NOTE: It is necessary to configure the broker similarly to achieve the desired behaviour with the image. An out-of-the-box broker.xml for a regular new broker instance contains various settings that differ from the code defaults used by this image out of the box "

Reply:
I don't agree. you notion of out of the box and regular, there is no such thing in general and if we want to containerise a broker, we need to configure it to customise it. The whole point of a container is to be immutable in so far as possible, so it is deterministic. To that end, a container builder should know what they want to achieve and configure as appropriate for sure. 
The whole point is to provide a base image that is extensible and as a base image, to be handy for simple demos or to play with. But  nothing more.

Reply:
There is clearly a thing such as out of the box behaviour. Its what you get when you start out, when you havent yet gone through every configuration option (which are not yet documented), or found what its effective-default is (again with the doc..), and set everything up to your precise needs. E.g, your readme even specifically makes the point that the produced image is ready to use as-is, and goes on to demonstrate exactly that.

I am not arguing against anything being extensible. I am not arguing it needs to behave the same.

I am saying that it should be clearer that it does not behave the same, as either the regular non-image broker or the artemis-docker images (which do behave the same as the regular non-image broker).  I would say it is _still_ not made that clear, so I really dont see how a user would actually expect such differences and handle them. It took me ages to notice this, even though I had already reviewed the not-yet-landed doc PR around broker properties and commented on how it also still doesnt make effective defaults clear for users of the properties.

I think we will need to agree to disagree here. I'll leave further review to other folks.